docs/feature-requests/005-editable-diff.md

```diff
@@ -0,0 +1,173 @@
+# Editable Diff
+
+## :memo: Summary
+
+Inline editing within a diff view (i.e. `MultiFilePatchView`).
+
+## :checkered_flag: Motivation
+
+This can save users the trouble of needing to toggle to an editor and back again when they notice typos, `console.log()` statements, or `.only()` in tests when reviewing changes right before committing.
+
+This also serves as a building block of the bigger [PR review workflow](https://github.com/atom/github/blob/master/docs/feature-requests/003-pull-request-review.md) we want to eventually implement.
+
+## 🤯 Explanation
+
+### What is editable?
+
+Currently, we use diff view in several places; which ones of them are editable?
+  - Unstaged Changes: *editable*
+  - Staged Changes: *editable* but with [unresolved questions](#question-unresolved-questions)
+  - All staged changes (aka Commit Preview): *editable* but with [unresolved questions](#question-unresolved-questions)
+  - Commit Detail Item: *NOT editable*
+  - Changed File Tab in PR: *editable **only** if it's a checked out PR*, also with [unresolved questions](#question-unresolved-questions)
+
+### Workflow of editing a diff
+
+#### 1. The entry point of editable state
+
+- By default, a diff view is editable (with some [exceptions](#what-is-editable) listed above). By clicking into any diff, a blinking caret will appear to signify users they can start editing.
+  ![editable](https://user-images.githubusercontent.com/378023/50885378-e71e0200-1430-11e9-98ca-6cab06b26bb4.gif)
+- If a diff is read-only, the caret is not visible. Selecting still works.
```

Comment:
When chatting with @donokuda, the following came up: When clicking on a read-only diff, no caret is shown, but when you start to type, a tooltip informs the user that this diff can't be edited. For PRs, it could say something like: "To edit, checkout this pull request".